monkey/common/agent_events/abstract_agent_event.py

```diff
@@ -24,7 +24,7 @@ class AbstractAgentEvent(InfectionMonkeyBaseModel, ABC):
         :param tags: The set of tags associated with the event
     """
 
-    source: AgentID
+    source: AgentID = Field(default=None)
```

Comment:
Makes no sense. If the default is `None`, then type hint should be at least `Optional[AgentID]`

Reply:
I will change it. I have stumbled upon issues because all of these fields needs to have defaults.

Reply:
Why do they all need defaults?

**THE SOURCE FIELD SHOULD NEVER BE NONE!!!**

There can never be an event that didn't originate somewhere.